lib/decanter/parser/boolean_parser.rb

```diff
@@ -5,9 +5,23 @@ class BooleanParser < ValueParser
       allow TrueClass, FalseClass
 
       parser do |val, options|
-        raise Decanter::ParseError.new 'Expects a single value' if val.is_a? Array
-        next if (val.nil? || val === '')
-        [1, '1'].include?(val) || !!/true/i.match(val.to_s)
+        normalized_val = normalize(val)
+        next if normalized_val.nil?
+
+        true_values = ['1', 'true']
+
+        option_val = options.fetch(:true_value, nil)
+        normalized_option = normalize(option_val)
+
+        true_values << normalized_option if normalized_option
+        true_values.find {|tv| !!/#{tv}/i.match(normalized_val)}.present?
```

Comment:
🍔 (food for thought): This can be refactored a bit to deal directly with booleans.

```rb
true_values.any? { |tv| /#{tv}/i.match?(normalized_val) }
```
https://ruby-doc.org/core-2.5.1/Regexp.html#method-i-match-3F
https://ruby-doc.org/core-2.7.0/Array.html#method-i-any-3F

Reply:
Seeing this code has made me realize that there is a defect here (which at this point would probably be considered a breaking change): we aren't matching the _entire_ string (i.e., looking for an exact match) 😱 .

This means,
```rb
!!/1/.match("false1") # true
!!/true/.match("this is not true") # true
```
I'll raise an issue for this separately.

Reply:
https://github.com/LaunchPadLab/decanter/issues/135

Reply:
good catch